aspnetcore/fundamentals/metapackage-app.md

```diff
@@ -0,0 +1,62 @@
+
+All the features of ASP.NET Core 2.1 and later and Entity Framework Core 2.1 and later are included in the `Microsoft.AspNetCore.App` package. The default project templates targeting ASP.NET Core 2.1 and later use this package. We recommend applications targeting ASP.NET Core 2.1+ and Entity Framework Core 2.1+ use the `Microsoft.AspNetCore.App` package.
+
+The version number of the `Microsoft.AspNetCore.App` metapackage represents the ASP.NET Core version and Entity Framework Core version. The ASP.NET Core version is currently aligned with the .NET Core version, and we hope to continue version alignment in the future.
+
+Using the the `Microsoft.AspNetCore.App` metapackage provides version restrictions that protect your app:
+
+* Other packages added to your app cannot change the version of packages included in `Microsoft.AspNetCore.App`.
+* Version consistency ensures a reliable experience. `Microsoft.AspNetCore.App` was designed to prevent untested version combinations of related bits being used together in the same app.
+
+Applications that use the `Microsoft.AspNetCore.App` metapackage automatically take advantage of the ASP.NET Core shared framework. When you use the `Microsoft.AspNetCore.App` metapackage, **no** assets from the referenced ASP.NET Core NuGet packages are deployed with the application &mdash; the ASP.NET Core shared framework contains these assets. The assets in the shared framework are precompiled to improve application startup time. For more information, see "shared framework" in [.NET Core distribution packaging](/dotnet/core/build/distribution-packaging).
+
+Packages in the .NET Core shared framework are excluded in published application output.
+
+The following *.csproj* file references the `Microsoft.AspNetCore.App` metapackage for ASP.NET Core:
+
+```xml
+<Project Sdk="Microsoft.NET.Sdk.Web">
+
+  <PropertyGroup>
+    <TargetFramework>netcoreapp2.1</TargetFramework>
+  </PropertyGroup>
+
+  <ItemGroup>
+    <PackageReference Include="Microsoft.AspNetCore.App" />
+  </ItemGroup>
+
+</Project>
+
+```
+
+The preceding markup doesn't specify a version number for the `Microsoft.AspNetCore.App` NuGet package. The ASP.NET Core 2.1+ templates generate this markup without a version number. When the version is not specified, the version used depends on the package source. On servers and in Azure, the package source is generally the ASP.NET Core shared framework. On development machines, it's generally `https://api.nuget.org/v3/index.json`.
+
+We recommend not specifying the version number. By not specifying the version, the shared framework roll forward mechanism will run the app on the latest version of the installed shared framework. To guarantee the same version is used in development, test, and production, ensure the same version of the shared framework is installed in all environments.
```

Comment:
Sorry I think my phrasing in the previous comment was confusing. To make this statement accurate, remove the "By not specifying the version," part.

The version is implicitly specified by the SDK, i.e. Microsoft.NET.Sdk.Web. However the implicit version is usually set to major.minor.0 for portable apps. The framework roll forward mechanism of the dotnet host chooses the most appropriate shared framework out of all those installed and runs the app on that. In most cases this means the latest installed major.minor.patch where the major.minor matches the implicit version number set by the SDK.

There are a few more nuances but I don't think it's necessary to go into all of those details. But if you want to take a look: https://github.com/dotnet/core-setup/blob/master/Documentation/design-docs/roll-forward-on-no-candidate-fx.md. Also, the implicit version number is set to the latest bundled major.minor.patch of the shared framework for standalone apps and this is handled through the SDK as well.